### PR TITLE
feat: persist session prompt defaults across bridge and mobile

### PR DESCRIPTION
**Pull request by AlexDochioiu**
## Summary
- add the shared `SessionPromptDefaults` model and thread session prompt defaults through shared session serialization
- persist prompt defaults in bridge Drift/storage flows, enrich session reads, and update defaults after successful create/send operations
- restore persisted prompt defaults in mobile session detail loading and picker state when the stored agent/provider/model/variant is still valid
- include `4f39f0bc test(mobile): sync session detail fixtures` from Task 7 so app session-detail tests match the required `currentSession` snapshot/listSessions behavior
- include `8bbf1c51 test(mobile): remove duplicate session list ui test` from Task 7 to remove the misplaced Flutter UI test from `mobile/module_core`

## Verification
- `shared/sesori_shared`: `env -u DART_ROOT dart run build_runner build --delete-conflicting-outputs` ✅, `env -u DART_ROOT dart test` ✅, `env -u DART_ROOT dart analyze` ✅
- `bridge`: `env -u DART_ROOT make codegen` ✅, `env -u DART_ROOT make test` ✅, `env -u DART_ROOT make analyze` ✅
- `mobile/app`: `env -u DART_ROOT flutter test` ✅ after fixture sync commit `4f39f0bc`
- `mobile/module_core`: `env -u DART_ROOT dart test` ✅ and `env -u DART_ROOT dart analyze` ✅ after removing the duplicate misplaced UI test in `8bbf1c51`
- `GIT_MASTER=1 git diff --check` ✅
- Aristotle implementation review ✅ after removing the duplicate `mobile/module_core/test/cubits/session_list/session_lifecycle_ui_test.dart`

## Notes
- the earlier pure-Dart runner caveat is resolved on this branch; `mobile/module_core` no longer contains the duplicate Flutter/theme UI test that caused the compile failure

---
## Summary by cubic
Persist and sync session prompt defaults (agent and model) across bridge and mobile. Defaults are stored in the bridge DB (schema v6), exposed via `POST /session/detail` and SSE, and used by mobile to prefill pickers and live-update when they change.

- **New Features**
  - Shared: Added `SessionPromptDefaults` to `Session` in `@sesori_shared` and a new SSE event `session.prompt_defaults_changed`.
  - Bridge: Session payloads now include `promptDefaults`. After a successful prompt send, stored defaults are updated and `session.prompt_defaults_changed` is emitted. Added `POST /session/detail` to fetch a single enriched session; project ID is resolved from stored session when available.
  - Mobile: Loads `promptDefaults` via `getSession` in the detail flow to restore picker state, listens for `session.prompt_defaults_changed` to refresh, uses the resolved project when loading providers, and fixes session ID extraction for this SSE event.

- **Migration**
  - Bridge DB migrates from v5 to v6 (adds nullable `lastAgent` and `lastAgentModel`). No manual steps; existing rows default to null.

<sup>Written for commit 904e55c4daaafc83e75ac23bd055f2fc0787476e. Summary will update on new commits. <a href="https://cubic.dev/pr/sesori-ai/sesori_apps_monorepo/pull/172?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: bridge/app/lib/src/bridge/persistence/daos/session_dao.dart [hunk 1]
```diff
@@ -18,6 +18,10 @@ class SessionDao extends DatabaseAccessor<AppDatabase> with _$SessionDaoMixin {
     required String? branchName,
     required String? baseBranch,
     required String? baseCommit,
+    String? lastAgent,
+    String? lastProviderId,
+    String? lastModelId,
+    String? lastVariantId,
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/persistence/daos/session_dao.dart]: Let's use the AgentModel class instead which includes provider id, model id and variant
```suggestion
    required String? lastAgent,
    required AgentModel? lastAgentModel,
```

File: bridge/app/lib/src/bridge/persistence/daos/session_dao.dart [hunk 2]
```diff
@@ -29,11 +33,32 @@ class SessionDao extends DatabaseAccessor<AppDatabase> with _$SessionDaoMixin {
...
       ),
     );
   }
 
+  Future<void> updatePromptDefaults({
+    required String sessionId,
+    required String? agent,
+    required String? providerId,
+    required String? modelId,
+    required String? variantId,
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/persistence/daos/session_dao.dart]:
```suggestion
    required String? agent,
    required AgentModel? agentModel
```

File: bridge/app/lib/src/bridge/persistence/database.dart [hunk 3]
```diff
@@ -108,6 +108,12 @@ class AppDatabase extends _$AppDatabase {
           );
         }
       },
+      from5To6: (m, schema) async {
+        await m.addColumn(schema.sessionsTable, schema.sessionsTable.lastAgent);
+        await m.addColumn(schema.sessionsTable, schema.sessionsTable.lastProviderId);
+        await m.addColumn(schema.sessionsTable, schema.sessionsTable.lastModelId);
+        await m.addColumn(schema.sessionsTable, schema.sessionsTable.lastVariantId);
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/persistence/database.dart]:
```suggestion
        await m.addColumn(schema.sessionsTable, schema.sessionsTable.lastAgentModel);
```

File: bridge/app/lib/src/bridge/repositories/session_repository.dart [hunk 4]
```diff
@@ -259,6 +259,56 @@ class SessionRepository {
...
+      );
+    });
+  }
+
+  Future<void> updatePromptDefaults({
+    required String sessionId,
+    required String? agent,
+    required String? providerId,
+    required String? modelId,
+    required String? variantId,
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/repositories/session_repository.dart]: use `AgentModel` class

File: bridge/app/lib/src/bridge/repositories/session_repository.dart [hunk 5]
```diff
@@ -259,6 +259,56 @@ class SessionRepository {
...
+    required bool isDedicated,
+    required int createdAt,
+    required String? worktreePath,
+    required String? branchName,
+    required String? baseBranch,
+    required String? baseCommit,
+    required String? agent,
+    required String? providerId,
+    required String? modelId,
+    required String? variantId,
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/repositories/session_repository.dart]: use `AgentModel` class

File: bridge/app/lib/src/bridge/persistence/tables/session_table.dart [hunk 6]
```diff
@@ -39,6 +43,10 @@ sealed class SessionDto with _$SessionDto, $SessionTableTableToColumns {
     required int? archivedAt,
     required String? baseBranch,
     required String? baseCommit,
+    required String? lastAgent,
+    required String? lastProviderId,
+    required String? lastModelId,
+    required String? lastVariantId,
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/persistence/tables/session_table.dart]:
```suggestion
    required AgentModel? lastAgentModel,
```

File: bridge/app/lib/src/bridge/persistence/tables/session_table.dart [hunk 7]
```diff
@@ -19,6 +19,10 @@ class SessionTable extends Table {
   IntColumn get archivedAt => integer().nullable()();
   TextColumn get baseBranch => text().nullable()();
   TextColumn get baseCommit => text().nullable()();
+  TextColumn get lastAgent => text().nullable()();
+  TextColumn get lastProviderId => text().nullable()();
+  TextColumn get lastModelId => text().nullable()();
+  TextColumn get lastVariantId => text().nullable()();
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/persistence/tables/session_table.dart]: Change to a single shared column for those 3 + a converter.
```suggestion
  TextColumn get lastAgentModel => text().map(const AgentModelConverter())();
```

An example converter is here:
```dart
class ColorConverter extends TypeConverter<Color, int> {
  const ColorConverter();

  @override
  Color fromSql(int fromDb) => Color(fromDb);

  @override
  // ignore: deprecated_member_use
  int toSql(Color value) => value.value;
}
```

You need a similar one. Just encode to string as `{providerId}|{modelId}|{variant}`.

File: shared/sesori_shared/lib/src/models/sesori/session.dart [hunk 8]
```diff
@@ -39,12 +40,25 @@ sealed class Session with _$Session {
     required SessionTime? time,
     required SessionSummary? summary,
     required PullRequestInfo? pullRequest,
+    SessionPromptDefaults? promptDefaults,
```
Comment (AlexDochioiu) [inline shared/sesori_shared/lib/src/models/sesori/session.dart]:
```suggestion
    required SessionPromptDefaults? promptDefaults,
```

File: shared/sesori_shared/lib/src/models/sesori/session.dart [hunk 9]
```diff
@@ -39,12 +40,25 @@ sealed class Session with _$Session {
...
   factory Session.fromJson(Map<String, dynamic> json) => _$SessionFromJson(json);
 }
 
+@Freezed(fromJson: true, toJson: true)
+sealed class SessionPromptDefaults with _$SessionPromptDefaults {
+  const factory SessionPromptDefaults({
+    required String? agent,
+    required String? providerID,
+    required String? modelID,
+    required String? variantID,
```
Comment (AlexDochioiu) [inline shared/sesori_shared/lib/src/models/sesori/session.dart]:
```suggestion
    required AgentModel? model,
```

File: mobile/module_core/lib/src/cubits/session_detail/session_detail_cubit.dart [hunk 10]
```diff
@@ -320,91 +308,15 @@ class SessionDetailCubit extends Cubit<SessionDetailState> {
...
-  ///
-  /// Conservative: if we can't tell at buffer time whether an event is
-  /// relevant (e.g. [SesoriSessionStatus] or [SesoriSessionUpdated] may be
-  /// for one of our child sessions), we queue it and let the replay handler
-  /// decide.
-  ///
-  /// [SesoriSessionsUpdated] is intentionally excluded: it triggers a silent
-  /// refresh, but during loading we are already fetching the latest snapshot,
-  /// so replaying it would cause a redundant refresh immediately after load.
-  bool _isRelevantGlobalEvent(SseEvent event) {
```
Comment (AlexDochioiu) [inline mobile/module_core/lib/src/cubits/session_detail/session_detail_cubit.dart:344]: Pretty sure you literally reverted here the work from another PR that got merged to main. My guess is that you screwed up during your rebase?

I am pretty sure this feature (entire PR) needs no changes in `mobile` at all. This can be fully a bridge-change, can't it?

File: bridge/app/lib/src/bridge/routing/get_session_handler.dart [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    required Map<String, String> pathParams,
+    required Map<String, String> queryParams,
+    required String? fragment,
+  }) async {
+    final sessionId = body.sessionId;
+    if (sessionId.isEmpty) {
+      throw buildErrorResponse(request, 400, "empty session id");
+    }
+
+    final projectId = await _sessionRepository.findProjectIdForSession(sessionId: sessionId);
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/routing/get_session_handler.dart:30]: Can we first check in our DB if we have project id for this session? before asking the plugin?

This change should be an implementation detail for `SessionRepository` directly.

File: bridge/app/lib/src/bridge/services/session_prompt_service.dart [hunk 12]
```diff
@@ -1,11 +1,18 @@
...
+import "../sse/sse_manager.dart";
 
 class SessionPromptService {
   final SessionRepository _sessionRepository;
+  final SSEManager? _sseManager;
 
-  SessionPromptService({required SessionRepository sessionRepository}) : _sessionRepository = sessionRepository;
+  SessionPromptService({
+    required SessionRepository sessionRepository,
+    SSEManager? sseManager,
```
Comment (AlexDochioiu) [inline bridge/app/lib/src/bridge/services/session_prompt_service.dart]:
```suggestion
    required SSEManager sseManager,
```
